src/app.ts

```diff
@@ -13,14 +13,37 @@ interface TemperatureSummary {
   average: number
 }
 
+let readingsGlobal: TemperatureReading[]
 export function processReadings(readings: TemperatureReading[]): void {
   // add here your code
-}
 
+  readingsGlobal = readings
+  return
+}
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
   //add here your code
-  return null
+  const filter = readingsGlobal.filter(
+    (data) => data.city === city && data.time.getTime() === date.getTime(),
+  )
+  if (filter.length < 1) {
+    return null
+  }
+  const TemperatureSummary = {
+    first: filter[0].temperature,
+    last: filter.slice(-1)[0].temperature,
```

Comment:
Could you explain me what are you doing here?
```suggestion
    last: filter.slice(-1)[0].temperature,
```

Reply:
yes, I wanted the last item, so I sliced it and because it returns an array, I selected the index 0 of the array that returns and got the temperature  from it